src/mod2c_core/modl.c

```diff
@@ -48,232 +46,154 @@ THE POSSIBILITY OF SUCH DAMAGE.
-		*cp = '\0';
-		 if (chdir(cs) == 0) {
-			printf("current directory is \"%s\"\n", cs);
-			strcpy(argv[1], cp+1);
-		}
-	}
-	}		
+static void openfiles(char* given_filename, char* output_dir) {
+  char  output_filename [NRN_BUFSIZE];
+  char  input_filename  [NRN_BUFSIZE];
+  modprefix = strdup (given_filename);                      // we want to keep original string to open input file
+  char* first_ext_char = strrchr(modprefix, '.');           // find last '.' that delimit file name from extension
+
+  if(first_ext_char) *first_ext_char = '\0';                // effectively cut the extension from prefix if it exist in given_filename
+  if ((fin = fopen(given_filename, "r")) == (FILE *) 0) {   // first try to open given_filename
+    Sprintf(input_filename, "%s.mod", given_filename);      // if it dont work try to add ".mod" extension and retry
+    Sprintf(finname, "%s.mod", given_filename);             // finname is still a global variable, so we need to update it
+    if ((fin = fopen(input_filename, "r")) == (FILE *) 0) {
+      diag("Can't open input file: ", input_filename);
+    }
+  }
+  if (output_dir[0] != '\0') {
+      char* basename = strrchr(modprefix,'/');
+      Sprintf(output_filename, "%s/%s.c", output_dir, basename);
+  }
+  else
+    Sprintf(output_filename, "%s.c", modprefix);
+
+  if ((fcout = fopen(output_filename, "w")) == (FILE *) 0) {
+  diag("Can't create C file: ", output_filename);
+  }
+  Fprintf(stderr, "Translating %s into %s\n", input_filename, output_filename);
+}
 
-#endif
+int main(int argc, char** argv) {
+  int option        = -1;
+  int option_index  = 0;
+  char* output_dir = "";
+  while ( (option = getopt_long (argc, argv, ":vho:", long_options, &option_index)) != -1) {
+    switch (option) {
+      case 'v':
+        printf("%s\n", nmodl_version_);
+        exit(0);
+      
+      case 'o':
+        output_dir = strdup(optarg);
+        break;
+      
+      case 'h':
+        fprintf(stderr, "%s source to source compiler from NMODL to C files\n", argv[0]);
+        fprintf(stderr, "Usage: %s [options] Inputfile\n", argv[0]);
+        fprintf(stderr, "Options:\n");
+        fprintf(stderr, "\t-o | --outdir <OUTPUT_DIRECTORY>    directory where output files will be written\n");
+        fprintf(stderr, "\t-h | --help                         print this message\n");
+        fprintf(stderr, "\t-v | --version                      print version number\n");
+        exit(0);
+      
+      case ':':
+        fprintf(stderr, "%s: option '-%c' requires an argument\n", argv[0], optopt);
+        exit (-1);
+      
+      case '?':
+      default:
+        fprintf(stderr, "%s: invalid option `-%c' \n", argv[0], optopt);
+        exit (-1);
+    }
+  }
+  if ((argc - optind) > 1) {
+    fprintf(stderr, "%s: Warning several input files specified on command line but only one will be processed\n", argv[0]);
+  }
+
+  init(); /* keywords into symbol table, initialize lists, etc. */
+  finname = argv[optind];
```

Comment:
this will segfaul (later in the code) when user don't pass anything : 

```
$ ./src/mod2c_core/mod2c_core
Segmentation fault: 11
```

we should keep old behaviour i.e. in case of no arguments display message:

```
$ ./src/mod2c_core/mod2c_core
Usage:modl prefixto.mod [inputfile]
```